packages/matchbox/src/components/TextField/TextField.js

```diff
@@ -1,179 +1,207 @@
-      this.calculatePadding();
-    }
+import { Box } from '../Box';
+import { HelpText } from '../HelpText';
+import { getRectFor, roundToBaseline } from '../../helpers/geometry';
+import useInputDescribedBy from '../../hooks/useInputDescribedBy';
+import { tokens } from '@sparkpost/design-tokens';
+import { margin } from 'styled-system';
+import styled from 'styled-components';
+import { pick, omit } from '@styled-system/props';
+
+const StyledWrapper = styled('div')`
+  ${margin}
+`;
+
+// TODO Make this reusable and shared with Select
+const FieldBox = props => {
+  return (
+    <Box
+      as="input"
+      disabled={props.disabled}
+      width="100%"
+      border={props.hasError ? `1px solid ${tokens.color_red_700}` : '400'}
+      borderRadius="100"
+      bg={props.disabled ? 'gray.200' : 'white'}
+      pl="400"
+      pr="600"
+      lineHeight="2.5rem"
+      height="2.5rem"
+      color="gray.900"
+      {...props}
+    />
+  );
+};
+
+function PrefixOrSuffix({ content, className, forwardedRef, ...rest }) {
+  if (!content) {
+    return null;
   }
 
-  calculatePadding() {
-    this.setState({
-      paddingLeft: this.prefixNode && this.prefixNode.getBoundingClientRect().width + 20,
-      paddingRight: this.suffixNode && this.suffixNode.getBoundingClientRect().width + 20
-    });
-  }
-
-  render() {
-    const {
-      align,
-      id,
-      name,
-      multiline,
-      type,
-      placeholder,
-      autoFocus,
-      disabled,
-      readOnly,
-      required,
-      resize,
-      label,
-      labelHidden,
-      helpText,
-      error,
-      errorInLabel,
-      onChange,
-      onFocus,
-      onBlur,
-      connectLeft,
-      connectRight,
-      prefix,
-      prefixClassname,
-      suffix,
-      suffixClassname,
-      style,
-      ...rest
-    } = this.props;
-
-    const { paddingLeft, paddingRight } = this.state;
-
-    const setInputClasses = classnames(
-      styles.Input,
-      error && styles.Error
-    );
-
-    const prefixClasses = classnames(styles.Prefix, prefixClassname);
-    const suffixClasses = classnames(styles.Suffix, suffixClassname);
+  return (
+    <Box
+      as="span"
+      position="absolute"
+      zIndex="1"
+      top="0"
+      lineHeight="2.5rem"
+      height="2.5rem"
+      className={className}
+      ref={forwardedRef}
+      {...rest}
+    >
+      {content}
+    </Box>
+  );
+}
 
-    const requiredIndicator = required
-      ? ' *'
-      : '';
+function TextField(props) {
+  const {
+    align,
+    autoFocus,
+    connectLeft,
+    connectRight,
+    disabled,
+    error,
+    errorInLabel,
+    helpText,
+    id,
+    label,
+    labelHidden,
+    multiline,
+    name,
+    onChange,
+    onFocus,
+    onBlur,
+    placeholder,
+    prefix,
+    prefixClassname,
+    readOnly,
+    required,
+    resize = 'both',
+    type,
+    suffix,
+    suffixClassname,
+    style,
+    ...rest
+  } = props;
+
+  const componentProps = omit(rest);
```

Comment:
Will this not potentially introduce bugs? We take advantage of prop spreading in the app a good bit.

Reply:
Hmm yea let me revert this

Reply:
Actually I think I have to leave this one in. Otherwise any `margin` system props passed onto the wrapper will be duplicated on the input